hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveCollectionPersister.java

```diff
@@ -0,0 +1,65 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.persister.collection.impl;
+
+import java.io.Serializable;
+import java.util.concurrent.CompletionStage;
+
+import org.hibernate.collection.spi.PersistentCollection;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+
+/**
+ * Similar to {@link org.hibernate.persister.collection.AbstractCollectionPersister} in ORM
+ * <p>
+ *     The reactive classes already extend something so we cannot have an abstract class like in ORM.
+ * </p>
+ */
+public interface ReactiveCollectionPersister {
+
+	/**
+	 * Reactive version of {@link org.hibernate.persister.collection.AbstractCollectionPersister}.recreate()
+	 * @param collection
+	 * @param id
+	 * @param session
+	 * @return Void completion stage
+	 */
+	CompletionStage<Void> recreateReactive(PersistentCollection collection, Serializable id, SharedSessionContractImplementor session);
+
+	/**
+	 * Reactive version of {@link org.hibernate.persister.collection.AbstractCollectionPersister}.remove()
+	 * @param id
+	 * @param session
+	 * @return Integer completion stage
+	 */
+	CompletionStage<Integer> removeReactive(Serializable id, SharedSessionContractImplementor session);
```

Comment:
This is the only method to return `CompletionStage<Integer>` instead of `CompletionStage<Void>` and it's also inconsistent with ORM.

Reply:
k, I'll work through the changes

Reply:
fixed and pushed